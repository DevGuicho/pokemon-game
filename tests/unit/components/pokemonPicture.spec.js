import { shallowMount, mount } from "@vue/test-utils";
import PokemonPicture from "@/components/PokemonPicture.vue";

describe("PokemonPicture component", () => {
  test("debe renderizar el componente correctamente", () => {
    const wrapper = mount(PokemonPicture, {
      props: { pokemonId: 100, showPokemon: false },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
  test("debe mostrar la imagen oculta y el pokemon 100", () => {
    const wrapper = mount(PokemonPicture, {
      props: { pokemonId: 100, showPokemon: false },
    });
    const [img1, img2] = wrapper.findAll("img");

    expect(img1.exists()).toBe(true);
    expect(img2).toBe(undefined);
    expect(img1.classes("hidden-pokemon")).toBe(true);
    expect(img1.attributes().src).toBe(
      "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg"
    );
  });
  test("debe mostrar el pokemon si showPokemon: true", () => {
    const wrapper = mount(PokemonPicture, {
      props: { pokemonId: 100, showPokemon: true },
    });
    const [img1, img2] = wrapper.findAll("img");

    expect(img1.exists()).toBe(true);
    expect(img2).toBe(undefined);
    expect(img1.classes("fade-in")).toBe(true);
  });
});
