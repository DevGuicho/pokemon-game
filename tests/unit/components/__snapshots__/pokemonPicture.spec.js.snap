// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPicture component debe renderizar el componente correctamente 1`] = `<div class="pokemon-container"><img class="hidden-pokemon" src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg" alt="pokemon"></div>`;
